skbio/io/format/gff3.py

```diff
@@ -0,0 +1,469 @@
+
+
+@gff3.writer(IntervalMetadata)
+def _interval_metadata_to_gff3(obj, fh, seq_id, skip=True):
+    '''
+    Parameters
+    ----------
+    obj : IntervalMetadata
+    seq_id : str
+        ID for column 1 in the GFF3 file.
+    '''
+    _serialize_interval_metadata(obj, seq_id, fh, skip=True)
+
+
+def _yield_record(fh):
+    '''Yield lines that belong to the same sequence.'''
+    lines = []
+    current = False
+    for line in _line_generator(fh, skip_blanks=True, strip=True):
+        if line.startswith('##FASTA'):
+            # stop once reaching to sequence section
+            break
+        if not line.startswith('#'):
+            try:
+                seq_id, _ = line.split('\t', 1)
+            except ValueError:
+                raise GFF3FormatError(
+                    'Wrong GFF3 format at line: %s' % line)
+            if current == seq_id:
+                lines.append(line)
+            else:
+                if current is not False:
+                    yield current, lines
+                lines = [line]
+                current = seq_id
+    yield current, lines
+
+
+def _parse_record(lines, interval_metadata):
+    '''Parse the lines into a IntervalMetadata object.'''
+    for line in lines:
+        columns = line.split('\t')
+        # there should be 9 columns
+        if len(columns) != 9:
+            raise GFF3FormatError(
+                'do not have 9 columns in this line: "%s"' % line)
+        # the 1st column is seq ID for every feature. don't store
+        # this repetitive information
+        metadata = {'source': columns[1],
+                    'type': columns[2],
+                    'score': columns[5],
+                    'strand': columns[6]}
+        phase = columns[7]
+        try:
+            metadata['phase'] = int(phase)
+        except ValueError:
+            if phase != '.':
+                raise GFF3FormatError(
+                    'unknown value for phase column: {!r}'.format(phase))
+        metadata.update(_parse_attr(columns[8]))
+
+        start, end = columns[3:5]
+
+        bounds = [(int(start)-1, int(end))]
+
+        interval_metadata.add(bounds, metadata=metadata)
+
+    return interval_metadata
+
+
+def _parse_attr(s):
+    '''parse attribute column'''
+    voca_change = _vocabulary_change('gff3')
+    md = {}
+    # in case the line ending with ';', strip it.
+    s = s.rstrip(';')
+    for attr in s.split(';'):
+        k, v = attr.split('=')
+        if k in voca_change:
+            k = voca_change[k]
+        md[k] = v
+    return md
+
+
+def _serialize_interval_metadata(
+        interval_metadata, seq_id, fh, skip=True):
+    '''Serialize an IntervalMetadata to GFF3.
+
+    Parameters
+    ----------
+    interval_metadata : IntervalMetadata
+    skip : bool
+        whether to skip outputting each sub region as a line in GFF3.
+    '''
+    # write file header
+    print('##gff-version 3', file=fh)
```

Comment:
Oh, I didn't see this was printing to a filehandle. Recommend using `fh.write()` since that's typically the API used to write to a filehandle (I didn't even know there was a `file` parameter with `print`!).